feature/interests/src/main/java/com/google/samples/apps/nowinandroid/feature/interests/InterestsRoute.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.nowinandroid.feature.interests
+
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+
+@Composable
+internal fun InterestsRoute(
+    listState: LazyListState,
+    shouldShowTwoPane: Boolean,
+    onTopicClick: (String) -> Unit,
+    onBackClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    viewModel: InterestsViewModel = hiltViewModel(),
+) {
+    val interestUiState by viewModel.interestUiState.collectAsStateWithLifecycle()
+    val topicUiState by viewModel.topicUiState.collectAsStateWithLifecycle()
+
+    Row(modifier = modifier) {
+        if (shouldShowTwoPane || topicUiState == null) {
```

Comment:
Is it better to add another state in TopicUiState to express a state where topic hasn't chose than expressing it as null because it's more descriptive?

Reply:
IMO, at the moment, the solution including null value is a bit more readable and straightforward. Once we introduce a placeholder for empty topic selection (as @alexvanyo suggested) I think it would make perfect sense to add an empty state. So if thats ok with you, I would like to leave it as nullable for now.

Reply:
Yup, that sounds good. Thanks for the clarification!